pkg/authorization/api/validation/validation.go

```diff
@@ -35,7 +35,14 @@ func ValidateSubjectAccessReview(review *authorizationapi.SubjectAccessReview) f
 	if len(review.Action.Verb) == 0 {
 		allErrs = append(allErrs, field.Required(field.NewPath("verb"), ""))
 	}
-	if len(review.Action.Resource) == 0 {
+	if review.Action.IsNonResourceURL {
+		if len(review.Action.Path) == 0 {
+			allErrs = append(allErrs, field.Required(field.NewPath("path"), ""))
+		}
+		if len(review.Action.Resource) != 0 {
```

Comment:
disallow all the resource-related fields (namespace, group, resource, resourcename) for non-resource urls